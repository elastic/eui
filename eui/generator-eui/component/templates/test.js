import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  test('is rendered', () => {
    const component = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
