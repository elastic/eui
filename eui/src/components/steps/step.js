import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  EuiScreenReaderOnly,
} from '../accessibility';

import {
  EuiTitle,
} from '../title';

import {
  STATUS,
  EuiStepNumber,
} from './step_number';

export const EuiStep = ({
  className,
  children,
  headingElement,
  step,
  title,
  status,
  ...rest
}) => {
  const classes = classNames('euiStep', className);

  let screenReaderPrefix;
  if (status === 'incomplete') {
    screenReaderPrefix = 'Incomplete';
  }

  return (
    <div
      className={classes}
      {...rest}
    >

      <EuiScreenReaderOnly><span>{screenReaderPrefix} Step </span></EuiScreenReaderOnly>

      <EuiStepNumber className="euiStep__circle" number={step} status={status} isHollow={status === 'incomplete'}/>

      <EuiTitle size="s" className="euiStep__title">
        {React.createElement(headingElement, null, title)}
      </EuiTitle>

      <div className="euiStep__content">
        {children}
      </div>

    </div>
  );
};

EuiStep.propTypes = {
  children: PropTypes.node.isRequired,
  /**
   * Will replace the number provided in props.step with alternate styling.
   * Options: `complete`, `incomplete`, `warning`, `danger`, `disabled`
   */
  status: PropTypes.oneOf(STATUS),
  /**
   * The number of the step in the list of steps
   */
  step: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  /**
   * The HTML tag used for the title
   */
  headingElement: PropTypes.string.isRequired,
};

EuiStep.defaultProps = {
  headingElement: 'p'
};
