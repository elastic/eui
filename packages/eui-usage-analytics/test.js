/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const Codeowners = require('codeowners');
const temp = new Codeowners("../../../kibana");

const { scan } = require('./scan');

const runScan = async () => {
  const scanResult = await scan();
  console.log(scanResult);
};

runScan();
