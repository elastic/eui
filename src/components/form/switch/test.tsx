import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
