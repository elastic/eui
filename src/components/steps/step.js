import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  EuiScreenReaderOnly,
} from '../accessibility';

import {
  EuiTitle,
} from '../title';

export const EuiStep = ({
  className,
  children,
  headingElement,
  step,
  title,
  ...rest
}) => {
  const classes = classNames('euiStep', className);
  return (
    <div
      className={classes}
      {...rest}
    >

      <EuiScreenReaderOnly><span>Step</span></EuiScreenReaderOnly>

      <EuiTitle className="euiStep__title" data-step-num={step}>
        {React.createElement(headingElement, null, title)}
      </EuiTitle>

      <div className="euiStep__content">
        {children}
      </div>

    </div>
  );
};

EuiStep.propTypes = {
  children: PropTypes.node.isRequired,
  step: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  headingElement: PropTypes.string.isRequired,
};

EuiStep.defaultProps = {
  headingElement: 'p'
};
