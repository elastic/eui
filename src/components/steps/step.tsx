import React, { FunctionComponent, HTMLAttributes, ReactNode } from 'react';
import { CommonProps } from '../common';

import classNames from 'classnames';

import { EuiTitle } from '../title';

import { EuiStepStatus, EuiStepNumber } from './step_number';

import { EuiI18n } from '../i18n';

export interface EuiStepProps {
  children: ReactNode;
  /**
   * The HTML tag used for the title
   */
  headingElement?: string;
  /**
   * The number of the step in the list of steps
   */
  step?: number;
  title: string;
  /**
   * May replace the number provided in props.step with alternate styling.
   */
  status?: EuiStepStatus;
}

export type StandaloneEuiStepProps = CommonProps &
  HTMLAttributes<HTMLDivElement> &
  EuiStepProps;

export const EuiStep: FunctionComponent<StandaloneEuiStepProps> = ({
  className,
  children,
  headingElement = 'p',
  step = 1,
  title,
  status,
  ...rest
}) => {
  const classes = classNames('euiStep', className);

  return (
    <div className={classes} {...rest}>
      <div className="euiStep__titleWrapper">
        <EuiI18n
          token="euiStep.ariaLabel"
          default={({ status }: { status?: EuiStepStatus }) => {
            if (status === 'incomplete') return 'Incomplete Step';
            return 'Step';
          }}
          values={{ status }}>
          {(ariaLabel: string) => (
            <EuiStepNumber
              className="euiStep__circle"
              aria-label={`${ariaLabel} ${step}`}
              number={step}
              status={status}
              isHollow={status === 'incomplete'}
            />
          )}
        </EuiI18n>

        <EuiTitle size="s" className="euiStep__title">
          {React.createElement(headingElement, null, title)}
        </EuiTitle>
      </div>

      <div className="euiStep__content">{children}</div>
    </div>
  );
};
