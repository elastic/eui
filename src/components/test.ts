export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
