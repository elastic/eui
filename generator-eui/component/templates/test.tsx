/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  it('renders', () => {
    const { container } = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
