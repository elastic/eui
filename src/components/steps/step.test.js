import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiStep } from './step';

describe('EuiStep', () => {
  test('is rendered', () => {
    const stepContent = (<p>Do this</p>);
    const component = render(
      <EuiStep
        {...requiredProps}
        headingElement={'h3'}
        step={1}
        title={'First step'}
        children={stepContent}
      />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
