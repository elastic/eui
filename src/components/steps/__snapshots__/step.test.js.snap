// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiStep is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiStep testClass1 testClass2"
  data-test-subj="test subject string"
>
  <span
    class="euiScreenReaderOnly"
  >
    Step
  </span>
  <h3
    class="euiTitle euiStep__title"
    data-step-num="1"
  >
    First step
  </h3>
  <div
    class="euiStep__content"
  >
    <p>
      Do this
    </p>
  </div>
</div>
`;
